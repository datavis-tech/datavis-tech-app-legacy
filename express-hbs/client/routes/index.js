var update = require('./update');
var read = require('./read');

var routes = {
  'update': update,
  'read': read
};

function start(route, connection, dataBundle){
  routes[route](connection, dataBundle);
}

module.exports = { start: start };
