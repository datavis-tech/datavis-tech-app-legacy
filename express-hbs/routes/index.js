var express = require('express');

var home = require('./home');
var libs = require('./libs');
var create = require('./create');
var read = require('./read');
var update = require('./update');

module.exports = function (connection){
  var router = express.Router();

  router.get('/', home);
  router.use('/libs', libs);
  router.post('/create', create(connection));
  router.get('/:id', read(connection));
  router.get('/:id/edit', update(connection));

  return router;
};
