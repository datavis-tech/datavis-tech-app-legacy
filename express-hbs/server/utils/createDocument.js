// This module provides a function that creates a new
// document in ShareDB.
function createDocument(connection, id, data, callback){

  data.content = [
    '<!DOCTYPE html>',
    '<html>',
    '  <head>',
    '    <meta charset="utf-8">',
    '    <meta name="viewport" content="width=device-width">',
    '    <title>Example</title>',
    '    <script src="https://d3js.org/d3.v4.min.js"></script>',
    '  </head>',
    '  <body>',
    '    <script>',
    '      d3.select("body")',
    '        .append("h1")',
    '          .text("Hello World!");',
    '    </script>',
    '  </body>',
    '</html>'
  ].join('\n');

  var doc = connection.get('documents', id);
  doc.create(data, callback);
}

module.exports = createDocument;
