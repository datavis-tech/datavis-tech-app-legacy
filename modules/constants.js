const AUTH_PATH = '/auth'
const AUTH_PATH_GITHUB = AUTH_PATH + '/github'
const AUTH_PATH_LOGOUT = AUTH_PATH + '/logout'
const AUTH_SUCCESS_REDIRECT = '/'
const AUTH_FAILURE_REDIRECT = AUTH_PATH + '/failed'
const AUTH_LOGOUT_REDIRECT = AUTH_SUCCESS_REDIRECT

const DB_DOCUMENTS_COLLECTION = 'documents'
const DB_USERS_COLLECTION = 'users'

module.exports = {
  AUTH_PATH,
  AUTH_PATH_GITHUB,
  AUTH_PATH_LOGOUT,
  AUTH_SUCCESS_REDIRECT,
  AUTH_FAILURE_REDIRECT,
  AUTH_LOGOUT_REDIRECT,
  DB_DOCUMENTS_COLLECTION,
  DB_USERS_COLLECTION
}
