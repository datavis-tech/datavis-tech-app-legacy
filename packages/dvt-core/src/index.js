const constants = require('./constants')
const db = require('./db')

module.exports = {
    constants,
    db
}
