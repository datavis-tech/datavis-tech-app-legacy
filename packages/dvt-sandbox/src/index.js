const flow = require('lodash/flow')
const partialRight = require('lodash/partialRight')
const get = require('lodash/get')

const { serializeDocument, accessors: { files: filesAccessor } } = require('dvt-core').db

const getDocument = require('./getDocument')
const getReferencedDocuments = require('./getReferencedDocuments')
const magicSandbox = require('./magicSandbox')
const { injectLogo, withBodyTag, withHtmlTag } = require('./html')

module.exports = (backend) => {

  const connection = backend.connect()

  const getDocumentFromConnection = getDocument(connection)
  const getReferencedDocumentsFromConnection = getReferencedDocuments(connection)

  const prepareHtml = flow([withHtmlTag, withBodyTag])
  const extractReferences = partialRight(get, 'references')
  const extractContent = partialRight(get, 'content')

  return async ({id, origin, withLogo = true, href = 'https://datavis.com'} = {}) => {

    const injectLogoWithHrefToVis = partialRight(injectLogo, href)
    const getProccessedHtml = flow([prepareHtml, injectLogoWithHrefToVis])
    const getMagicSandbox = partialRight(magicSandbox, origin)

    const serializedDocument = serializeDocument(await getDocumentFromConnection(id))
    const serializedReferencedDocuments = (await getReferencedDocumentsFromConnection(serializedDocument.referencesIds)).map(serializeDocument)

    const getDocumentFilesFromReferences = partialRight(filesAccessor, serializedReferencedDocuments)

    const getFiles = flow([extractReferences, getDocumentFilesFromReferences])
    const getContent = flow([extractContent, getProccessedHtml])

    return getMagicSandbox(getContent(serializedDocument), getFiles(serializedDocument))
  }
}