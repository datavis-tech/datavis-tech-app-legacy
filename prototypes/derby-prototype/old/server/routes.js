var express = require("express");
var emailer = require("./emailer");

var router = express.Router();

// When an RFP is submitted,
// email it to curran@datavis.tech.
router.post("/rfp", function(req, res){
  var to = "curran@datavis.tech";
  var d = req.body;

  var mailOptions = {
    from: emailer.noreply,
    to: to,
    subject: "RFP from " + d.email,
    html: [
      "<h1>" + d.title + "</h1>",
      "<p>" + d.description + "</h1>",
      d.data ? "<h2>Sample Data</h2><pre>" + d.data + "</pre>" : ""
    ].join("")
  };

  emailer.sendMail(mailOptions, function(err, info){
    if(err){
      console.log(err);
    } else {
      console.log("Message sent: " + info.response);
    }
  });
  res.redirect("/rfpsubmitted");
});

module.exports = router;
