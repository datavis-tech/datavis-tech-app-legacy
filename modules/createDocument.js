import connection from './shareDBConnection'
import uuidV4 from 'uuid/v4'
import { DB_DOCUMENTS_COLLECTION } from './constants'

const generateId = () => uuidV4().replace(/-/g, '')

export default ({ title, description, owner }) => {
  const id = generateId()
  const doc = connection.get(DB_DOCUMENTS_COLLECTION, id)

  doc.create({
    title,
    description,
    owner
    // files
  })

  return id
}
