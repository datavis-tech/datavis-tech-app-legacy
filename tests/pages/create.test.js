import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import nodeSelector from '../utils/nodeSelector'
import { VIS_DOC_TYPE, DATA_DOC_TYPE } from '../../src/constants'
import { Form } from 'semantic-ui-react'
import fakeUser from '../utils/fakeUser'
import fakeDoc from '../utils/fakeDoc'

const mockUser = fakeUser()
const mockDoc = fakeDoc()
const mockTitle = 'This is a Title'

jest.mock('../../src/components/page', () => Page => Page)
jest.mock('../../src/components/layout', () => ({children}) => children)
jest.mock('../../src/routes', () => ({ Router: { pushRoute: jest.fn() } }))
jest.mock('../../src/db/actions/createDocument', () => ({ createDocument: jest.fn(() => mockDoc) }))

import CreatePage from '../../src/pages/create'
import { createDocument } from '../../src/db/actions/createDocument'
import { Router } from '../../src/routes'

describe('create page', () => {

  it('should invoke createDocument for vis type', async () => {
    const sut = mount(<CreatePage type={VIS_DOC_TYPE} user={mockUser} />)
    sut.find(nodeSelector('title-input')).instance().value = mockTitle

    createDocument.mockClear()
    expect(sut.find(Form).props().onSubmit(new Event('test')))
    expect(createDocument).toHaveBeenCalledWith({
      title: mockTitle,
      owner: mockUser.id,
      type: VIS_DOC_TYPE
    })
  })

  it('should invoke createDocument for data type', async () => {
    const sut = mount(<CreatePage type={DATA_DOC_TYPE} user={mockUser} />)
    sut.find(nodeSelector('title-input')).instance().value = mockTitle

    createDocument.mockClear()
    expect(sut.find(Form).props().onSubmit(new Event('test')))
    expect(createDocument).toHaveBeenCalledWith({
      title: mockTitle,
      owner: mockUser.id,
      type: DATA_DOC_TYPE
    })
  })

  it('should invoke pushRoute to edit page with new doc id', async () => {
    const sut = mount(<CreatePage type={VIS_DOC_TYPE} user={fakeUser()} />)
    Router.pushRoute.mockClear()
    expect(sut.find(Form).props().onSubmit(new Event('test')))
    expect(Router.pushRoute).toHaveBeenCalledWith('edit', { id: mockDoc.id })
  })

  it('should render correct layout', () => {
    const tree = renderer
      .create(<CreatePage type={DATA_DOC_TYPE} user={mockUser} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
