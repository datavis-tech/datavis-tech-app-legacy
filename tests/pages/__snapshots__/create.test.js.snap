// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create page should render correct layout 1`] = `
<div
  className="ui text container"
>
  <h1
    data-test="create-heading"
  >
    Create a 
    Dataset
     
    <i
      aria-hidden="true"
      className="info circle icon"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </h1>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <label>
        Title
      </label>
      <input
        data-test="title-input"
      />
    </div>
    <button
      className="ui primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Create
    </button>
  </form>
</div>
`;
